lib/src/main/java/uk/gov/di/ipv/core/library/validation/TokenRequestValidator.java

```diff
@@ -40,59 +40,59 @@ public TokenRequestValidator(ConfigurationService configurationService) {
     public void authenticateClient(String requestBody, Map<String, String> queryParams)
             throws ClientAuthenticationException {
         if (!queryParams.containsKey(CLIENT_ASSERTION_PARAM)) {
-            if (!queryParams.containsKey(CLIENT_ID_PARAM)) {
-                LOGGER.error(
-                        "Missing either client_assertion or client_id values in request. Failed to establish client_id value.");
-                throw new ClientAuthenticationException(
-                        "Unknown client, no client_id value or client_assertion jwt found in request");
-            } else {
-                clientId = queryParams.get(CLIENT_ID_PARAM);
-
-                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
-
-                if (clientAuthenticationMethod.equals(JWT)) {
-                    LOGGER.error("Missing client_assertion jwt for configured client {}", clientId);
-                    throw new ClientAuthenticationException(
-                            String.format(
-                                    "Missing client_assertion jwt for configured client '%s'",
-                                    clientId));
-                }
-            }
+            authenticateClientWithoutJwt(queryParams);
         } else {
-            PrivateKeyJWT clientJwt;
-            try {
-                clientJwt = PrivateKeyJWT.parse(requestBody);
-
-                clientId = clientJwt.getClientID().getValue();
+            authenticateClientWithJwt(requestBody);
+        }
+    }
 
-                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
+    private void authenticateClientWithoutJwt(Map<String, String> queryParams)
+            throws ClientAuthenticationException {
+        if (!queryParams.containsKey(CLIENT_ID_PARAM)) {
```

Comment:
Again, could we use `if (queryParams.containsKey(CLIENT_ID_PARAM))` and switch the blocks?